### PR TITLE
Require JobOffersExport and nil verification

### PR DESCRIPTION
**Pull request by corentindms**
Two small things in this pull request :
- It adds a require to `dr_rockter/job_offers_export` in the main file
- The deserializer now returns nil for a date attribute that is `null` in a JSON

### CODE REVIEW

File: lib/dr_rockter.rb [hunk 1]
```diff
@@ -1,4 +1,5 @@
 require "dr_rockter/version"
+require "dr_rockter/job_offers_export"
```
Comment (r3trofitted) [inline lib/dr_rockter.rb]: Could you elaborate on the reasoning behind this `require`?

As a general rule, I tend to prefer libraries that require as few modules and classes by default, and let users choose what to explicitly require, instead of early-loading everything. Do you think that this file should constitute an exception?

File: lib/dr_rockter/md.rb [hunk 2]
```diff
@@ -3,15 +3,15 @@
 
 module DrRockter
   DeserializerError = Class.new(StandardError)
-  
+
```
Comment (r3trofitted) [inline lib/dr_rockter/md.rb]: I'm not against stripping extra whitespaces (though I **do** like them), but could you do it in a separate PR so as not to pollute this one?

File: lib/dr_rockter/md.rb [hunk 3]
```diff
@@ -3,15 +3,15 @@
...
   module MD
     KNOWN_TYPES = {
       string:   ->(v) { v },
-      datetime: ->(v) { DateTime.parse v },
+      datetime: ->(v) { DateTime.parse v unless v.nil? },
```
Comment (r3trofitted) [inline lib/dr_rockter/md.rb:10]: 👍

**Review by r3trofitted — CHANGES_REQUESTED** [review]

File: lib/dr_rockter.rb (hunk 1 above)
Comment (corentindms) [reply lib/dr_rockter.rb]: Looking at popular gems, this file is generally used to require all the other files exposed by the gem, the dependencies, add some helper functions etc ... as it is automatically added to the load path by rubygem 
This allows the user to only require this file which have the gem name, and to not care about the file structure of the gem.

In a Rails app, it allows us to not require any file at all.

The goal of this require was to expose the `JobOffersExport` object without the need to require it. But it was maybe not the best way.

We have two choices :
Either I remove it, and we keep it the way you intended to (which is to let the users require the file that they need)
Or we can require all the files that we wish to expose.

If we choose the second option, we can ask ourselves if we want to require all the dependencies that we need to, or only require them in the files that needs them.

File: lib/dr_rockter/md.rb (hunk 2 above)
Comment (corentindms) [reply lib/dr_rockter/md.rb]: 👍

File: lib/dr_rockter.rb (hunk 1 above)
Comment (r3trofitted) [reply lib/dr_rockter.rb]: I'd rather not require everything, all the time, for the time being. Global `require`'s can  easily hide coupling, and this is not something that I'd like to encourage. Once this library is a bit more mature, we should have a better understanding of, as you say, the files that we wish to expose.

**Review by r3trofitted — APPROVED** [review]
:shipit:
